docs/website/docs/dlt-ecosystem/verified-sources/rest_api.md

```diff
@@ -0,0 +1,527 @@
+
+Let's break down the configuration in more detail.
+
+### Configuration structure
+
+:::tip
+Import the `RESTAPIConfig` type from the `rest_api` module to have convenient hints in your editor/IDE:
+
+```py
+from rest_api import RESTAPIConfig
+```
+:::
+
+
+The configuration object passed to the REST API Generic Source has three main elements:
+
+```py
+config: RESTAPIConfig = {
+    "client": {
+        ...
+    },
+    "resource_defaults": {
+        ...
+    },
+    "resources": [
+        ...
+    ],
+}
+```
+
+#### `client`
+
+`client` contains the configuration to connect to the API's endpoints. It includes the following fields:
+
+- `base_url` (str): The base URL of the API. This string is prepended to all endpoint paths. For example, if the base URL is `https://api.example.com/v1/`, and the endpoint path is `users`, the full URL will be `https://api.example.com/v1/users`.
+- `headers` (dict, optional): Additional headers to be sent with each request.
+- `auth` (optional): Authentication configuration. It can be a simple token, a `AuthConfigBase` object, or a more complex authentication method.
+- `paginator` (optional): Configuration for the default pagination to be used for resources that support pagination. See the [pagination](#pagination) section for more details.
+
+#### `resource_defaults` (optional)
+
+`resource_defaults` contains the default values to configure the dlt resources. This configuration is applied to all resources unless overridden by the resource-specific configuration.
+
+For example, you can set the primary key, write disposition, and other default settings here:
+
+```py
+config = {
+    "client": {
+        ...
+    },
+    "resource_defaults": {
+        "primary_key": "id",
+        "write_disposition": "merge",
+        "endpoint": {
+            "params": {
+                "per_page": 100,
+            },
+        },
+    },
+    "resources": [
+        "resource1",
+        "resource2": {
+            "name": "resource2_name",
+            "write_disposition": "append",
+            "endpoint": {
+                "params": {
+                    "param1": "value1",
+                },
+            },
+        },
+    ],
+}
+```
+
+Above, all resources will have `primary_key` set to `id`, `resource1` will have `write_disposition` set to `merge`, and `resource2` will override the default `write_disposition` with `append`.
+Both `resource1` and `resource2` will have the `per_page` parameter set to 100.
+
+#### `resources`
+
+This is a list of resource configurations that define the API endpoints to be loaded. Each resource configuration can be:
+- a dictionary with the [resource configuration](#resource-configuration).
+- a string. In this case, the string is used as the both as the endpoint path and the resource name, and the resource configuration is taken from the `resource_defaults` configuration if it exists.
+
+### Resource configuration
+
+A resource configuration has the following fields:
```

Comment:
here you should refer that on top of `endpoint` and `include_from_parent` we accept the same "hints" the `dlt.resource` decorator accepts. so there are way more of them than those below

Reply:
@rudolfix yes, it'd be nice to just refer the resource with all its possible params to not duplicate the content. It would be better to link to a resource object which has a list of possible hints. 
Should I link https://deploy-preview-1308--dlt-hub-docs.netlify.app/docs/api_reference/extract/decorators#resource ?

Reply:
Fixed in [a3e1d43](https://github.com/dlt-hub/dlt/pull/1308/commits/a3e1d438ddba215ad313d4ad59ca3307db15d6b1)